sbt-scalajs-callgraph/src/main/scala/ch/epfl/sbtplugin/CallGraphPlugin.scala

```diff
@@ -24,46 +23,42 @@ object CallGraphPlugin extends AutoPlugin {
   import ScalaJSPlugin.autoImport._
   import autoImport._
 
-  override lazy val projectSettings: Seq[Setting[_]] = {
-    Seq(
-      callgraph := {
-        val log = streams.value.log
-        val ir = (scalaJSIR in Compile).value.data
-        val linker = (scalaJSLinker in Compile).value
-        val outputMode = (scalaJSOutputMode in Compile).value
-        val withSourceMap = (emitSourceMaps in Compile).value
-        val semantics = linker.semantics
-        val symbolRequirements =
-          new BasicLinkerBackend(semantics, outputMode, withSourceMap, LinkerBackend.Config())
-            .symbolRequirements
-
-        Try {
-          linker.linkUnit(ir, symbolRequirements, log)
-        } match {
-          case Success(linkUnit) =>
-            val infos = linkUnit.infos.values.toSeq
-            createCallGraph(infos)
-          case Failure(e) =>
-            log.warn(e.getMessage)
-            log.warn("Non linking program, falling back to all the *.sjsir files on the classpath...")
-            val infos = ir map (_.info)
-            createCallGraph(infos)
-        }
 
-        def createCallGraph(infos: Seq[Infos.ClassInfo]) = {
-          val analysis = Analyzer.computeReachability(semantics, symbolRequirements, infos, false)
-          val graph = Graph.createFrom(analysis)
+  lazy val configSettings: Seq[Setting[_]] = Seq(
+    callgraph := {
+      val log = streams.value.log
+      val ir = scalaJSIR.value.data
+      val linker = scalaJSLinker.value
+      val outputMode = scalaJSOutputMode.value
+      val withSourceMap = emitSourceMaps.value
+      val semantics = linker.semantics
+      val symbolRequirements =
+        new BasicLinkerBackend(semantics, outputMode, withSourceMap, LinkerBackend.Config())
+          .symbolRequirements
 
-          val jsonFile = crossTarget.value / "graph.json"
-          Graph.writeToFile(graph, jsonFile)
-          log.info(s"CallGraph file created in $jsonFile")
+      val infos = Try {
+        linker.linkUnit(ir, symbolRequirements, log)
+      } match {
+        case Success(linkUnit) =>
+          linkUnit.infos.values.toSeq
+        case Failure(e) =>
+          log.warn(e.getMessage)
+          log.warn("Non linking program, falling back to all the *.sjsir files on the classpath...")
+          ir map (_.info)
+      }
 
-          val htmlFile = crossTarget.value / "index.html"
-          HTMLFile.writeToFile(htmlFile)
-          log.info(s"HTML file created in $htmlFile")
-        }
+      val analysis = Analyzer.computeReachability(semantics, symbolRequirements, infos, false)
+      val graph = Graph.createFrom(analysis)
+      val file = artifactPath.value
+      HTMLFile.writeToFile(file, graph, false)
+      log.info(s"HTML file created in $file")
+    },
+    artifactPath in Compile := crossTarget.value / "callgraph.html",
+    artifactPath in Test := crossTarget.value / "callgraph-test.html"
```

Comment:
Er ... nope ^^ That will conflict with any other plugin that would reuse `artifactPath` unscoped per task.
For `artifactPath` should define `artifactPath in (Compile, callgraph) := ...`, and use it as `(artifactPath in callgraph).value` inside `callgraph :=`.